### PR TITLE
Update labextension build

### LINKED ISSUE #354: Outdated build system for the jupyterlab extension and failing to update yarn.lock
I've struggled to do chore maintenance with the jupyterlab extension. It started by me seing a note about its a good idea to remove yarn.lock to recreate it with new dependencies before relaese. Doing that, things have broken by missing typescript stuff.

I'm not confident on how to do the typical maintenance task associated with the JupyterLab extension.

Is there someone who knows a good reference example to model from with regards to the jupyterlab extension, some relevant documentation to link to, and/or is able to directly help with migrating this extensions possibly outdated stuff in package.json and yarn.lock?

## EDIT: Clarification

What fails is the following, which is fine if yarn.lock remain. The text is from #353 but the current RELEASE.md is describing the same things.

>## Steps to make a release
>
>1. Create a PR updating `jupyterlab-server-proxy/yarl.lock` and continue only
>   when its merged.
>
>   This helps us avoid leaving known vulnerabilities are unfixed. To do this,
>   delete the file and manually perform the the `build dist` step in the
>   `.github/workflows/publish.yaml` workflow's `build` job as summarized below.
>
>   ```shell
>   rm jupyterlab-server-proxy/yarn.lock
>
>   pip install jupyter_packaging wheel jupyterlab~=3.0
>   python setup.py sdist bdist_wheel
>   ```

### PR DESCRIPTION
**Pull request by bollwyvl**
## References

- fixes #354

## Motivation

Without changing too much, this updates some key versions and approaches. Going further would take rather a bit of doing, and is probably not going to buy much.

## Changes
- [x] update typescript
- [x] update build slightly to clean before doing a prod build 
- [x] update build instructions... these were mostly referencing jupyterlab 2-era things
- [x] copy LICENSE to the package 
- [x] resolve `yarn.lock`
- [x] further tighten up `yarn.lock` with `yarn-deduplicate`
- [x] add an explicit yarn cache, aka "offline mirror"

### CODE REVIEW

**Comment by bollwyvl** [comment]
Little bit of thrash: feel free to squash after review, but otherwise no further changes planned.

File: jupyterlab-server-proxy/README.md [hunk 1]
```diff
@@ -2,31 +2,30 @@
...
 
 ```bash
-npm install
-npm run build
-jupyter labextension link .
+conda install jupyter-server-proxy
 ```
 
-To rebuild the package and the JupyterLab app:
+As a _prebuilt_ extension, it will "just work," and requires not
```
Comment (consideRatio) [inline jupyterlab-server-proxy/README.md:23]: An incomplete sentence

File: jupyterlab-server-proxy/README.md [hunk 2]
```diff
@@ -2,31 +2,30 @@
 
 Launcher icons for proxied applications
 
-
 ## Prerequisites
 
-* JupyterLab
+- JupyterLab >=2,<3
```
Comment (consideRatio) [inline jupyterlab-server-proxy/README.md:7]: This confuses me significantly - as jupyterlab 3 support exists for this extension

**Review by consideRatio — APPROVED** [review]
Thank you @bollwyvl!!!! :heart: :tada: :sunflower: 

I raised documentation questions, but otherwise this LGTM!

**Comment by consideRatio** [comment]
I've just merged #353 that updates RELEASE.md and can conclude that this PR makes the instructions about updating yarn.lock works again!! Thank you @bollwyvl!!!

File: jupyterlab-server-proxy/README.md (hunk 2 above)
Comment (bollwyvl) [reply jupyterlab-server-proxy/README.md]: d9aff30

File: jupyterlab-server-proxy/README.md (hunk 1 above)
Comment (bollwyvl) [reply jupyterlab-server-proxy/README.md]: d9aff30

**Comment by bollwyvl** [comment]
Resolved the above docs conversations: feel free to reopen if i made further mistakes :blush:

**Comment by consideRatio** [comment]
:heart: :tada: thank you soo much @bollwyvl!!!

**Pull request merged**